docs/en/stack/security/securing-communications/tutorial-tls-addnodes.asciidoc

```diff
@@ -39,71 +18,117 @@ in a separate folder. You can simply repeat the steps that you used to install
-----
+You are prompted for information about each new node. Specify `node-2` and
+`node-3` for the instance names. For the purposes of this tutorial, specify the
+same IP address (`127.0.0.1,::1`) and DNS name (`localhost`) for each node.
 
-Add the following settings to the `ES_PATH_CONF/elasticsearch.yml`
+By default, the command produces a zip file named `certificate-bundle.zip`,
+which contains the generated certificates and keys.
+--
+
+. Decompress the `certificate-bundle.zip` file. For example:
++
+--
+["source","sh",subs="attributes,callouts"]
+----------------------------------------------------------------------
+unzip certificate-bundle.zip 
+
+Archive:  certificate-bundle.zip
+   creating: node-2/
+  inflating: node-2/node-2.p12       
+   creating: node-3/
+  inflating: node-3/node-3.p12   
+----------------------------------------------------------------------
+// NOTCONSOLE
+  
+The `certificate-bundle.zip` file contains a folder for each of your nodes. Each
+folder contains a single PKCS#12 keystore that includes a node certificate,
+node key, and CA certificate.
+--
+
+. Create a folder to contain certificates in the configuration directory of each
+{es} node. For example, create a `certs` folder in the `config` directory.
+
+. Copy the appropriate certificate to the configuration directory on each node.
+For example, copy the `node-2.p12` file into the `config/certs` directory on the
+second node and the `node-3.p12` into the `config/certs` directory on the third
+node.
+
+. Specify the name of the cluster and give each node a unique name.
++
+--
+For example, add the following settings to the `ES_PATH_CONF/elasticsearch.yml`
 file on the second node:
 
 [source,yaml]
 ----
-xpack.security.enabled: true
 cluster.name: test-cluster
 node.name: node-2
-discovery.zen.minimum_master_nodes: 2
-transport.tcp.port: 9302
-discovery.zen.ping.unicast.hosts: ["localhost:9301", "localhost:9303"]
 ----
 
 Add the following settings to the `ES_PATH_CONF/elasticsearch.yml`
 file on the third node:
 
 [source,yaml]
 ----
-xpack.security.enabled: true
 cluster.name: test-cluster
 node.name: node-3
-discovery.zen.minimum_master_nodes: 2
-transport.tcp.port: 9303
-discovery.zen.ping.unicast.hosts: ["localhost:9301", "localhost:9302"]
 ----
 
-TIP: In these examples, we have not specified the `transport.host`,
-`transport.bind_host`, or `transport.publish_host` settings, so they default to
-the `network.host` value. If you have not specified the `network.host` setting,
-it defaults to `_local_`, which represents the loopback addresses for the system. 
+NOTE: In order to join the same cluster as the first node, they must share the
+same `cluster.name` value.
+
+--
 
-If you choose different cluster names, node names, host names, or ports, you
-must substitute the appropriate values in subsequent steps as well. 
+. If you set the `cluster.initial_master_nodes` on the first node, specify the
```

Comment:
I think we should be positively recommending setting `cluster.initial_master_nodes` on all three of these nodes to make sure that they form a single cluster. It'll mostly work without that, as long as nothing runs too slowly, but it seems easy enough to be sure.

Reply:
Initially I didn't set it on the first node (and was therefore able to start the first node prior to adding the two others). Then since the cluster had been started once successfully, it seemed redundant to bother adding it to the other two (and stopping the first node to re-add it there).

Do you think it's worth adding in that scenario nonetheless?

Reply:
Ah I see, I hadn't realised we'd already formed a cluster by this point. We recommend not setting it on nodes joining an existing cluster at all, so it shouldn't be set here. However, we're not setting `discovery.seed_hosts` either, so auto-bootstrapping might happen. For instance if people try to adjust this tutorial to set up a cluster across nodes then it won't be obvious they need to do this. Can we set it on the first node and set `discovery.seed_hosts` on the other two? I think that'll help it translate to the multi-node case better.

Reply:
Thanks for the clarification!  I've update the tutorial to set cluster.initial_master_nodes on the first node and discovery.seed_hosts on the second and third node.
https://doc-review-63bf6.firebaseapp.com/x2/encrypting-internode.html
https://doc-review-63bf6.firebaseapp.com/x2/encrypting-communications-hosts.html